appservice/multiplexer.py

```diff
@@ -130,8 +134,23 @@ async def handle_session_new(request):
     logger.debug('new_session result status %i, content: %s', pod_status, content)
 
     if pod_status >= 200 and pod_status < 300:
-        response = JSONResponse({'id': sessionid})
-        await update_session(sessionid, 'wait_target')
+        session_hostname = f'session-{sessionid}{SESSION_INSTANCE_DOMAIN}'
+        # resolve and cache IP addresses now, to avoid DNS lag/trouble during proxying
+        for retry in range(30):
+            try:
+                addr = socket.gethostbyname(session_hostname)
```

Comment:
This is no good -- in podman this can take 30 s (!!) until it times out and raises a "Name or service not found". This needs to be moved into a background task.

Reply:
This is a podman bug. I filed https://github.com/containers/podman/issues/15972 about it. I'll ignore this for now, as it is specific to our testing and podman with netavark.